File: CS/11_SecurityAndSignatures/CustomSignature.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Security;

namespace CustomSignature
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void button1_Click(object sender, EventArgs e)
        {
            // Specify the input and output file paths
            string input = @"..\..\..\..\..\..\Data\DigitalSignature.pdf";
            string output = "addSelfDefineSignaturePic.pdf";

            // Load the PDF document from the disk
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(input);

            // Get the first page of the document
            PdfPageBase page = doc.Pages[0];

            // Load the certificate for the digital signature
            PdfCertificate cert = new PdfCertificate(@"..\..\..\..\..\..\Data\gary.pfx", "e-iceblue");

            // Create a new PdfSignature object with the document, page, certificate, and identifier
            PdfSignature signature = new PdfSignature(doc, page, cert, "demo");

            // Set the bounds (position and size) of the signature on the page
            signature.Bounds = new RectangleF(50, 600, 200, 200);

            // Configure custom graphics for the signature area using the DrawGraphics method defined below
            signature.ConfigureCustomGraphics(DrawGraphics);

            // Save the modified document to the specified output file path in PDF format
            doc.SaveToFile(output, FileFormat.PDF);

            PDFDocumentViewer(output);
        }
        // Method to draw custom graphics within the signature area
        private void DrawGraphics(PdfCanvas g)
        {
            // Create a PdfTrueTypeFont object with Arial font and size 18
            PdfTrueTypeFont font = new PdfTrueTypeFont(new Font("Arial", 18f));
            //////////////////Use the following code for netstandard dlls/////////////////////////
            /*
             PdfTrueTypeFont font = new PdfTrueTypeFont("Arial", 18f);
            */

            // Specify the text to be drawn
            string text = "Signature information";

            // Measure the height of the font
            float heightY = font.MeasureString(text).Height;

            // Define the starting point for drawing the string
            PointF point1 = new PointF(0, 0);

            // Draw the string using the specified font, brush color, and position
            g.DrawString(text, font, PdfBrushes.Red, point1);

            // Define the starting point for drawing the image
            PointF point2 = new PointF(0, heightY + 10);

            // Draw the image using the specified image file and position
            g.DrawImage(PdfImage.FromFile(@"..\..\..\..\..\..\Data\E-iceblueLogo.png"), point2);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/11_SecurityAndSignatures/CustomSignature.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics
Imports Spire.Pdf.Security

Namespace CustomSignature
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Load the input file path
            Dim input As String = "..\..\..\..\..\..\Data\DigitalSignature.pdf"

            ' Set the output file name
            Dim output As String = "addSelfDefineSignaturePic.pdf"

            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Load the PDF document from the input file path
            doc.LoadFromFile(input)

            ' Get the first page of the document
            Dim page As PdfPageBase = doc.Pages(0)

            ' Create a new PdfCertificate object with the certificate file path and password
            Dim cert As New PdfCertificate("..\..\..\..\..\..\Data\gary.pfx", "e-iceblue")

            ' Create a new PdfSignature object with the document, page, certificate, and signature name
            Dim signature As New PdfSignature(doc, page, cert, "demo")

            ' Set the bounds for the signature appearance on the page
            signature.Bounds = New RectangleF(50, 600, 200, 200)

            ' Configure custom graphics for the signature appearance using the DrawGraphics method
            signature.ConfigureCustomGraphics(AddressOf DrawGraphics)

            ' Save the modified document to the output file path in PDF format
            doc.SaveToFile(output, FileFormat.PDF)

            ' Close the PDF document
            doc.Close()

            ' Launch the document
            PDFDocumentViewer(output)
        End Sub
        Private Sub DrawGraphics(ByVal g As PdfCanvas)
            ' Create a new PdfTrueTypeFont object using the Arial font with a size of 18 points
            Dim font As New PdfTrueTypeFont(New Font("Arial", 18.0F))
            ' =============================================================================
            ' Use the following code for netstandard dlls
            ' =============================================================================
            'Dim font As New PdfTrueTypeFont("Arial", 18.0F, FontStyle.Regular, True)
            ' =============================================================================

            ' Specify the text to be drawn
            Dim text As String = "Signature information"

            ' Determine the height of the text using the font's MeasureString method
            Dim heightY As Single = font.MeasureString(text).Height

            ' Define the starting point for drawing the text as (0, 0)
            Dim point1 As New PointF(0, 0)

            ' Draw the specified text using the font, red color, and starting point
            g.DrawString(text, font, PdfBrushes.Red, point1)

            ' Define the starting point for drawing the image just below the drawn text
            Dim point2 As New PointF(0, heightY + 10)

            ' Draw the image from the specified file using the starting point
            g.DrawImage(PdfImage.FromFile("..\..\..\..\..\..\Data\E-iceblueLogo.png"), point2)
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace
